### PR TITLE
Add connectOrCreate example

### LINKED ISSUE #4415: Add Example of using connectOrCreate in Explicit Many to Many relations
In the Querying Explicit Many to Many relations [section](https://www.prisma.io/docs/concepts/components/prisma-schema/relations/many-to-many-relations#querying-an-explicit-many-to-many), there are examples of using nested create as well as connect, but there is no example of using connectOrCreate.

In the sample schema for explicit many-to-many relations, the `CategoriesOnPosts` model has a composite id that is a combination of `categoryId` and `postId`, it would be helpful to demonstrate an example that showcases passing of categoryId and postId for identifying a record uniquely.

```typescript
  await prisma.post.create({
    data: {
      title: "Hello Post",
      categories: {
        connectOrCreate: {
          where: {
            postId_categoryId: {
              postId: 1,
              categoryId: 1,
            },
          },
          create: {
            category: {
              create: {
                name: "Hello Category",
              },
            },
          },
        },
      },
    },
  });
``` 

A snippet similar to above can be added to highlight that a combination of postId and categoryId needs be passed in the where clause of connectOrCreate.

### PR DESCRIPTION
**Pull request by TasinIshmam**
Fixes #4415  

Adds an example with `connectOrCreate` in the Explicit Many to Many relations section.

### CODE REVIEW

File: content/200-concepts/100-components/01-prisma-schema/06-relations/300-many-to-many-relations.mdx [hunk 1]
```diff
@@ -90,7 +90,7 @@ The following section demonstrates how to query an explicit many-to-many relatio
 The following query:
 
 1. Creates a `Post`
-2. Creates a category assigment, or `CategoriesOnPosts` (assigned by Bob, assigned today)
+2. Creates a category assignment, or `CategoryToPost` (assigned by Bob, assigned today)
```
Comment (nikolasburk) [inline content/200-concepts/100-components/01-prisma-schema/06-relations/300-many-to-many-relations.mdx]: I find this phrasing not very clear, especially for a beginner. The "or `CategoryToPost` could be read as an _actual_ or (as in _either ... or ..._) but not as (what I assume is intended) representing a different phrasing or what a "category assignment" is on a technical level. Can you try to clarify this? 

(I hope my explanation makes sense, if no, feel free to ask for more clarification. Also, it seems that you just [reused existing phrasing](https://www.prisma.io/docs/concepts/components/prisma-schema/relations/many-to-many-relations#querying-an-explicit-many-to-many) so I'd ask you to update the original phrasing of this as well 🙏 )

**Review by nikolasburk — DISMISSED** [review]
looks great, one small change request  🙏

File: content/200-concepts/100-components/01-prisma-schema/06-relations/300-many-to-many-relations.mdx (hunk 1 above)
Comment (TasinIshmam) [reply content/200-concepts/100-components/01-prisma-schema/06-relations/300-many-to-many-relations.mdx]: I clarified what the relation table is and explicitly called out that `CategoriesOnPosts` as a relation table in each of the examples.

File: content/200-concepts/100-components/01-prisma-schema/06-relations/300-many-to-many-relations.mdx [hunk 2]
```diff
@@ -21,9 +21,9 @@ In relational databases, m-n-relations are typically modelled via [relation tabl
 In an explicit many-to-many relation, the **relation table is represented as a model in the Prisma schema** and can be used in queries. Explicit many-to-many relations define three models:
 
 - Two models that have a many-to-many relation, such as `Category` and `Post`
```
Comment (ruheni) [inline content/200-concepts/100-components/01-prisma-schema/06-relations/300-many-to-many-relations.mdx]:
```suggestion
- Two models with many-to-many relation, such as `Category` and `Post`.
```

**Pull request merged**